### PR TITLE
Add hotkeys on Pull Request tabs

### LINKED ISSUE #568: Add shortcuts to navigate to PR tabs
I think it would be useful to have keyboard shortcuts to navigate to Conversation, Commits and Files changed tabs in a pull request.

I think the keyboard shortcuts should start with <kbd>g</kbd> so they don't collide with other things or be unintuitive.

Do you support that so we could disucss specific shortcuts?

### PR DESCRIPTION
**Pull request by stivaugoin**
Closes #568 

<kbd>p</kbd> <kbd>1</kbd> Conversation
<kbd>p</kbd> <kbd>2</kbd> Commits
<kbd>p</kbd> <kbd>3</kbd> Files changed

<kbd>p</kbd> <kbd>←</kbd> Previous tab
<kbd>p</kbd> <kbd>→</kbd> Next tab

### CODE REVIEW

**Comment by bfred-it** [comment]
Thank you Steve!

A couple things before merging this:

- It needs to be merged in the readme
- The shortcuts (1,2,3) need to be added to the shortcuts help https://github.com/sindresorhus/refined-github/pull/1107

**Comment by bfred-it** [comment]
> Can we also support `p h` & `p l` for navigating between them?

Why?

**Comment by vanniktech** [comment]
I find them to be useful too.

**Comment by bfred-it** [comment]
> I find them to be useful too.

Why those specific keys? Everyone might have _useful_ keys but we should not support more than one combination.

**Comment by vanniktech** [comment]
Oh they come from the vim navigation and are pretty common. (Inbox, Vim, google Calendar etc)

File: source/features/add-pull-request-hotkey.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+	const selectedIndex = tabs.indexOf(select('.tabnav-pr .selected'));
+
+	for (const [index, tab] of tabs.entries()) {
+		const keys = [`p ${index + 1}`];
+		if (index === selectedIndex - 1) {
+			keys.push('p ArrowLeft');
+		} else if (index === selectedIndex + 1) {
+			keys.push('p ArrowRight');
+		}
+		tab.setAttribute('data-hotkey', keys);
```
Comment (sindresorhus) [inline source/features/add-pull-request-hotkey.js]: Use `.dataset`

**Comment by sindresorhus** [comment]
I think the arrow key shortcuts should cycle, so when you're at "Files changed" and press `=>` it should go to  "Conversation", and when at "Conversation" and pressing `<=`, it should go to "Files changed".

File: source/features/add-pull-request-hotkey.js (hunk 1 above)
Comment (bfred-it) [reply source/features/add-pull-request-hotkey.js]: Good idea. Also... https://github.com/github/eslint-plugin-github/blob/master/docs/rules/no-dataset.md

**Comment by sindresorhus** [comment]
I understand it's a limitation of the `hotkeys` attribute, but it's a bit annoying that you can not just hold `p` down and tap `arrow right` multiple times to go through the tabs. Now I have to press `p`, then `arrow right`, release, press `p`, then `arrow right`...

File: source/features/add-pull-request-hotkey.js (hunk 1 above)
Comment (sindresorhus) [reply source/features/add-pull-request-hotkey.js]: I don't think greppable should be a priority, and if you consistently use `dataset`, you can just do the conversion yourself when grepping.

Comment (sindresorhus) [reply source/features/add-pull-request-hotkey.js]: https://github.com/sindresorhus/eslint-plugin-unicorn/issues/166

**Comment by stivaugoin** [comment]
Hey guys, sorry i'm a little bit busy right now to finish this PR but I found the <kbd>p</kbd> hotkey is already assign in PR when the repo had Project activated.

github-project

**Comment by bfred-it** [comment]
Could it be `t` for _tab_ instead?

**Comment by stivaugoin** [comment]
> Could it be t for tab instead?

The <kbd>t</kbd> is already assign to `Jump to` in `Files changed` tab

**Comment by hkdobrev** [comment]
We could use <kbd>q</kbd> as closest to the digits on QWERTY keyboards and it's like "`Q`uick switcher".

**Comment by SuperSandro2000** [comment]
Q is commonly used a quit short cut

**Comment by stivaugoin** [comment]
What do you think of

<kbd>shift</kbd> + <kbd>←</kbd>
<kbd>shift</kbd> + <kbd>→</kbd>

🤔

**Comment by sindresorhus** [comment]
> Q is commonly used as quit short cut in a lot of Windows programs. It would be free but missleading.

What apps? Source? As long as it's not used in Chrome it should be fine to use it.

**Comment by SuperSandro2000** [comment]
It is free in chrome.

**Comment by vanniktech** [comment]
Q also sounds good to me.

**Comment by stivaugoin** [comment]
I can finish this PR this weekend. Do we take <kbd>q</kbd>?

**Comment by stivaugoin** [comment]
Hey guys, i'm not super comfortable to write in english, so feel free to correct my lines in README or completely rewrite it 😃

File: source/features/add-pull-request-hotkey.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import select from 'select-dom';
+import {registerShortcut} from './improve-shortcut-help';
+
+export default function () {
+	registerShortcut('pr', 'q 1', 'Go to Conversation');
+	registerShortcut('pr', 'q 2', 'Go to Commits');
+	registerShortcut('pr', 'q 3', 'Go to Files changed');
```
Comment (bfred-it) [inline source/features/add-pull-request-hotkey.js]: This will have to be updated

**Comment by bfred-it** [comment]
I think we can revert to `g`. We initially excluded `g` because it was hard to find reasonable _second letter_ but with numbers and arrows it's much easier.

How about:

<kbd>g</kbd> <kbd>1</kbd> 
<kbd>g</kbd> <kbd>2</kbd> 
<kbd>g</kbd> <kbd>3</kbd> 
<kbd>g</kbd> <kbd>4</kbd> 

and

<kbd>g</kbd> <kbd>←</kbd> 
<kbd>g</kbd> <kbd>→</kbd>

**Comment by hkdobrev** [comment]
@stivaugoin @sindresorhus Any thoughts on the updates?

**Comment by stivaugoin** [comment]
Sounds good to me 👍

File: source/content.js [hunk 3]
```diff
@@ -74,6 +74,7 @@ import monospaceTextareas from './features/monospace-textareas';
 import improveShortcutHelp from './features/improve-shortcut-help';
 import displayIssueSuggestions from './features/display-issue-suggestions';
 import addPullRequestHotkey from './features/add-pull-request-hotkey';
+import openSelectionInNewTab from './features/add-selection-in-new-tab';
```
Comment (stivaugoin) [inline source/content.js]: Is it necessary?

Comment (bfred-it) [reply source/content.js]: That’s part of the merge on master, not this PR

**Comment by hkdobrev** [comment]
🙌 🎉

**Comment by daxmc99** [comment]
Does this still work?

**Comment by yakov116** [comment]
It broke I will send a pr to fix it

**Comment by iamstarkov** [comment]
did it break again?

**Comment by fregante** [comment]
👆 it's better to open a new issue

**Pull request merged**